### PR TITLE
Add cuProj to docs.yml

### PR DESCRIPTION
**Pull request by jarmak-nv**
This PR adds cuProj to the docs.yml supporting its docs builds. It also updates the description of cuSpatial to a more up-to-date description.

Question: I set the legacy flag to `0` for cuProj, I wasn't sure if that was necessary but figured it was since there's no `23.06` version.

### CODE REVIEW

**Comment by raydouglass** [comment]
This PR depends on https://github.com/rapidsai/cuspatial/pull/1237

File: _data/docs.yml
Comment (bdice) [inline _data/docs.yml:75]: This link is a 404. Once the release PR is merged, this will be the stable link you want:
```suggestion
    ghlink: https://github.com/rapidsai/cuspatial/tree/main/python/cuproj
```

File: _data/docs.yml
Comment (bdice) [inline _data/docs.yml:201]:
```suggestion
    ghlink: https://github.com/rapidsai/cuspatial/tree/main/cpp/cuproj
```

**Comment by jarmak-nv** [comment]
TY @bdice I always forget about the `/tree/main` part of the URL when manually generating it.

**Comment by ajschmidt8** [comment]
The `23.10` docs for `cuproj` aren't built yet.

So we'll need to do one of the following before this PR is merged:

- set `cuproj`'s `nightly` value to `0` for this PR
- wait for this forward-merge to be resolved so the `cuproj` docs for `23.10` are built: https://github.com/rapidsai/cuspatial/pull/1244
  - there are open PRs for this, one of them will need to be merged:
    - https://github.com/rapidsai/cuspatial/pull/1246
    - https://github.com/rapidsai/cuspatial/pull/1251

**Review by jakirkham — APPROVED** [review]
Thanks Ben! 🙏

**Comment by ajschmidt8** [comment]
> The `23.10` docs for `cuproj` aren't built yet.


The `cuproj` docs for `23.10` are now available.

I think we should set `nightly` and `stable` to `1` for `cuproj` now and merge this after the `23.08` release PR.

**Review by bdice — APPROVED** [review]

**Pull request merged**
